snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ImportRequest.java

```diff
@@ -353,27 +360,52 @@ private DB createDb() {
 		}
 	}
 	
-	private void updateLocales(final BranchContext context, final ResourceURI codeSystemUri) throws Exception {
-		/*
-		 * XXX: The default language in locales is always "en", as there is no
-		 * machine-readable information about what language code each language type
-		 * reference set is associated with.
-		 */
-		final List<ExtendedLocale> locales = SnomedRequests.prepareSearchRefSet()
+	private void updateCodeSystemSettings(final BranchContext context, final ResourceURI codeSystemUri) throws Exception {
+		SnomedReferenceSets languageReferenceSets = SnomedRequests.prepareSearchRefSet()
 			.all()
 			.filterByType(SnomedRefSetType.LANGUAGE)
 			.filterByActive(true)
 			.setFields(SnomedConceptDocument.Fields.ID)
 			.sortBy(SortField.ascending(SnomedConceptDocument.Fields.ID))
 			.build()
-			.execute(context)
+			.execute(context);
+		
+		/*
+		 * XXX: The default language in locales is always "en", as there is no
```

Comment:
We could get the language code part from the dataset by looking at the language code of each description that is a member of the reference set in question. Something for the future.